### PR TITLE
Nostradamus

### PR DESCRIPTION
**Pull request by Topvennie**
New positioner (rebased on station-ws-conn)

Features:
- Supports switchovers
- Supports teams taking a break
- Is dialed in after about 2-3 laps
- No jumps

Requires a restart:
- New teams
- New stations

Todo (at a later stage):
- Smoothen out the speed when a new station is reached

### CODE REVIEW

File: src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import java.util.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class Nostradamus implements Positioner {
+    private static final Logger logger = Logger.getLogger(Nostradamus.class.getName());
+    private final int INTERVAL_CALCULATE_MS = 500; // How often to handle new detections (in milliseconds)
+    private final int INTERVAL_FETCH_MS = 60000; // Interval between fetching baton switchovers (in milliseconds)
```
Comment (redfast00) [inline src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java]: ik denk dat dit wel vaker mag gebeuren? of zorgt dat ergens voor problemen?

File: src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    private final int INTERVAL_DETECTIONS_MS = 3000; // Amount of milliseconds to group detections by
+    private final int MAX_NO_DATA_MS = 30000; // Send a stationary position after receiving no station update for x amount of milliseconds
+    private final int AVERAGE_AMOUNT = 10; // Calculate the median running speed of the last x intervals
```
Comment (redfast00) [inline src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java]: median or average?

File: src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java (hunk 1 above)
Comment (Topvennie) [reply src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java]: Shorten the interval [27d1dc4](https://github.com/12urenloop/Telraam/pull/142/commits/27d1dc4055b92d194436a531b12b30fe1d82eda1)

File: src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java (hunk 2 above)
Comment (Topvennie) [reply src/main/java/telraam/logic/positioner/nostradamus/Nostradamus.java]: It's the only reference left to average. All the 10 other references say median so it's median but changed it anyway [877fe72](https://github.com/12urenloop/Telraam/pull/142/commits/877fe72e94662b1b303f2f14819238420283cf84)

**Review by NuttyShrimp — APPROVED** [review]

**Pull request merged**